apps/mobile/src/features/accounts/components/account-card.tsx

```diff
@@ -1,68 +1,92 @@
-  name: string;
-  walletName: string;
+import { Box, Pressable, PressableProps, Text, usePressedState } from '@leather.io/ui/native';
+
+interface AccountCardProps extends PressableProps {
+  icon: AccountIcon | ComponentType;
+  caption?: string;
+  primaryTitle: ReactNode;
+  secondaryTitle?: ReactNode;
+  address?: ReactNode;
   onPress?(): void;
   onLongPress?(): void;
-  onLayout?(e: LayoutChangeEvent): void;
   testID?: string;
   iconTestID?: string;
 }
+
 export function AccountCard({
-  address,
-  balance,
+  caption,
+  primaryTitle,
+  secondaryTitle,
   icon,
-  name,
-  walletName,
+  address,
   onPress,
-  onLongPress,
-  onLayout,
-  testID,
-  iconTestID,
+  ...props
 }: AccountCardProps) {
-  const Container = onPress ? TouchableOpacity : Box;
+  const { pressed, onPressIn, onPressOut } = usePressedState(props);
+  const animatedStyle = useAnimatedStyle(() => ({
+    transform: [{ scale: withSpring(pressed ? 0.95 : 1) }],
+  }));
 
   return (
-    <Container
-      onPress={onPress}
-      onLongPress={onLongPress}
-      flexDirection="column"
-      p="5"
-      borderRadius="sm"
+    <Pressable
+      height={156}
+      p="4"
+      justifyContent="space-between"
+      bg="ink.background-primary"
       borderWidth={1}
-      borderColor="ink.border-transparent"
-      backgroundColor="ink.background-primary"
-      onLayout={onLayout}
-      testID={testID}
+      borderStyle="solid"
+      borderColor="ink.border-default"
+      borderRadius="md"
+      shadowOpacity={0.04}
+      shadowOffset={{ width: 0, height: 2 }}
+      shadowRadius={6}
+      onPress={onPress}
+      onPressIn={onPressIn}
+      onPressOut={onPressOut}
+      style={animatedStyle}
+      {...props}
     >
+      <Box flexDirection="row" justifyContent="space-between">
+        <AccountAvatar icon={icon} />
+        {address}
+      </Box>
       <Box
-        mb="6"
-        p="2"
-        alignSelf="flex-start"
-        borderRadius="round"
-        bg="ink.text-primary"
-        testID={iconTestID}
+        flexDirection="row"
+        justifyContent="space-between"
+        alignItems="flex-end"
+        gap="2"
+        mb="-0.5"
       >
-        {icon}
-      </Box>
-      <Box flexDirection="row" justifyContent="space-between">
-        <Box>
-          <Text variant="label01">{name}</Text>
-          <Text variant="caption01" color="ink.text-subdued">
-            {walletName}
-          </Text>
-        </Box>
-        <Box>
-          {balance}
-          {address}
+        <Box gap="0.5" flexShrink={1}>
+          {caption && (
+            <Text variant="label03" numberOfLines={1}>
+              {caption}
+            </Text>
+          )}
+          <Title>{primaryTitle}</Title>
         </Box>
+        <Title>{secondaryTitle}</Title>
       </Box>
-    </Container>
+    </Pressable>
+  );
+}
+
+function Title({ children }: HasChildren) {
+  if (!children) {
+    return null;
+  }
+
+  return isString(children) ? (
```

Comment:
slight preference for early return, more readable and consistent with above return